### PR TITLE
Update nrfX to version 1.0

### PR DESCRIPTION
**Pull request by sjanc**

### CODE REVIEW

**Comment by mlaz** [comment]
The changes look good to me. Although I'd suggest some testing which I can help with.

**Comment by sjanc** [comment]
Yeap, I did some testing on nrf52 (both 832 and 840) with BLE and things seem to be working fine.
If you have usecases you want to test please do, I'd like to get this merged soon so that it can be part of 1.4.

I'll also do nrf51 testing (blinky and ble) before merging.

File: hw/mcu/nordic/src/ext/nrfx/templates/nrfx_glue.h [hunk 1]
```diff
@@ -98,6 +105,27 @@ extern "C" {
  */
 #define NRFX_IRQ_DISABLE(irq_number)
 
+/**
+ * @brief Macro for setting a specific IRQ as pending.
+ *
+ * @param irq_number  IRQ number.
+ */
+#define NRFX_IRQ_PENDING_SET(irq_number)
```
Comment (mlaz) [inline hw/mcu/nordic/src/ext/nrfx/templates/nrfx_glue.h]: This and some other new macros should be updated(i.e. defined) on [mcu/nordic/include/nrfx_glue.h](https://github.com/apache/mynewt-core/blob/master/hw/mcu/nordic/include/nrfx_glue.h).

**Comment by sjanc** [comment]
@kasjer FYI

**Comment by jacobrosenthal** [comment]
Heh. they got 1.1 out today
https://github.com/NordicSemiconductor/nrfx/releases/tag/v1.1.0

**Comment by mlaz** [comment]
Looks good and drivers are now compiling. However I wonder whether we want v1.1.0

**Comment by sjanc** [comment]
Thanks for testing!
I'll update to 1.1 in separate PR later on

**Pull request merged**